pkg/scraper/scraper.go

```diff
@@ -83,110 +77,145 @@ func RegisterScraperMetrics(registrationFunc func(metrics.Registerable) error) e
 	return nil
 }
 
-func NewScraper(nodeLister v1listers.NodeLister, client client.KubeletMetricsGetter, scrapeTimeout time.Duration) *scraper {
-	return &scraper{
-		nodeLister:    nodeLister,
-		kubeletClient: client,
-		scrapeTimeout: scrapeTimeout,
+func NewManageNodeScraper(client client.KubeletMetricsGetter, scrapeTimeout time.Duration, metricResolution time.Duration, store storage.Storage) *manageNodeScraper {
+	return &manageNodeScraper{
+		kubeletClient:    client,
+		scrapeTimeout:    scrapeTimeout,
+		metricResolution: metricResolution,
+		stop:             map[apitypes.UID]chan struct{}{},
+		isWorking:        map[apitypes.UID]bool{},
+		storage:          store,
 	}
 }
 
-type scraper struct {
-	nodeLister    v1listers.NodeLister
-	kubeletClient client.KubeletMetricsGetter
-	scrapeTimeout time.Duration
+type manageNodeScraper struct {
+	nodeLock         sync.Mutex
+	kubeletClient    client.KubeletMetricsGetter
+	scrapeTimeout    time.Duration
+	metricResolution time.Duration
+	// Tracks all running per-node goroutines - per-node goroutine will be
+	// processing updates received through its corresponding channel.
+	stop map[apitypes.UID]chan struct{}
+	// Track the current state of per-node goroutines.
+	// Currently all update request for a given node coming when another
+	// update of this node is being processed are ignored.
+	isWorking map[apitypes.UID]bool
+
+	storage storage.Storage
 }
 
-var _ Scraper = (*scraper)(nil)
-
 // NodeInfo contains the information needed to identify and connect to a particular node
 // (node name and preferred address).
 type NodeInfo struct {
 	Name           string
 	ConnectAddress string
 }
 
-func (c *scraper) Scrape(baseCtx context.Context) *storage.MetricsBatch {
-	nodes, err := c.nodeLister.List(labels.Everything())
-	if err != nil {
-		// report the error and continue on in case of partial results
-		klog.ErrorS(err, "Failed to list nodes")
+func (m *manageNodeScraper) AddNodeScraper(node *corev1.Node) error {
+	m.nodeLock.Lock()
+	defer m.nodeLock.Unlock()
+	if working, exists := m.isWorking[node.UID]; exists && working {
+		klog.V(1).ErrorS(fmt.Errorf("Scrape in node is already running"), "", "node", klog.KObj(node))
```

Comment:
You should not call verbose `V(1)` with `ErrorS`. 

Let's change it to Info log.